src/lib/transition_router/transition_router.ml

```diff
@@ -149,26 +158,35 @@ module Make (Inputs : Inputs_intf) :
     let verified_transition_reader, verified_transition_writer =
       create_bufferred_pipe ()
     in
-    let frontier = Mvar.take_now_exn frontier_mvar in
     let ( transition_frontier_controller_reader
         , transition_frontier_controller_writer ) =
       start_transition_frontier_controller ~verified_transition_writer
-        ~clear_reader frontier
+        ~clear_reader
+        (Mvar.peek_exn frontier_mvar)
     in
     let mode =
```

Comment:
since this is the result of `Broadcaster.create`, should this variable be named `broadcaster` instead of `mode`?